src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -234,7 +235,21 @@ void ConsoleProcess::commonInit()
                            options_.smartTerminal ? core::system::kSmartTerm :
                                                     core::system::kDumbTerm);
 
+      // environment variables that can be used when configuring custom shells
       core::system::setenv(&(options_.environment.get()), "RSTUDIO_TERM", procInfo_->getHandle());
+
+      core::system::setenv(&(options_.environment.get()), "RSTUDIO_PROJ_NAME",
+                           projects::projectContext().file().stem());
+
+      FilePath projPath = core::system::userHomePath();
```

Comment:
Not sure I understand the goal of such a file, let's discuss realtime. In the meantime it looks like you've removed this path entirely (possibly in favor of just using a raw session id?) so will merge as-is and we can make future tweaks on master.